clients/mobile/ios/TodoList/Mobile.swift

```diff
@@ -335,12 +348,444 @@ fileprivate struct FfiConverterString: FfiConverter {
+    typealias SwiftType = TaskStatusDto
+
+    public static func read(from buf: inout (data: Data, offset: Data.Index)) throws -> TaskStatusDto {
+        let variant: Int32 = try readInt(&buf)
+        switch variant {
+        
+        case 1: return .created
+        
+        case 2: return .completed
+        
+        default: throw UniffiInternalError.unexpectedEnumCase
+        }
+    }
+
+    public static func write(_ value: TaskStatusDto, into buf: inout [UInt8]) {
+        switch value {
+        
+        
+        case .created:
+            writeInt(&buf, Int32(1))
+        
+        
+        case .completed:
+            writeInt(&buf, Int32(2))
+        
+        }
+    }
+}
+
+
+public func FfiConverterTypeTaskStatusDTO_lift(_ buf: RustBuffer) throws -> TaskStatusDto {
+    return try FfiConverterTypeTaskStatusDTO.lift(buf)
+}
+
+public func FfiConverterTypeTaskStatusDTO_lower(_ value: TaskStatusDto) -> RustBuffer {
+    return FfiConverterTypeTaskStatusDTO.lower(value)
+}
+
+
+extension TaskStatusDto: Equatable, Hashable {}
+
+
+
+fileprivate struct FfiConverterSequenceTypeTaskDTO: FfiConverterRustBuffer {
+    typealias SwiftType = [TaskDto]
+
+    public static func write(_ value: [TaskDto], into buf: inout [UInt8]) {
+        let len = Int32(value.count)
+        writeInt(&buf, len)
+        for item in value {
+            FfiConverterTypeTaskDTO.write(item, into: &buf)
+        }
+    }
+
+    public static func read(from buf: inout (data: Data, offset: Data.Index)) throws -> [TaskDto] {
+        let len: Int32 = try readInt(&buf)
+        var seq = [TaskDto]()
+        seq.reserveCapacity(Int(len))
+        for _ in 0 ..< len {
+            seq.append(try FfiConverterTypeTaskDTO.read(from: &buf))
+        }
+        return seq
+    }
+}
+private let UNIFFI_RUST_FUTURE_POLL_READY: Int8 = 0
+private let UNIFFI_RUST_FUTURE_POLL_MAYBE_READY: Int8 = 1
+
+fileprivate func uniffiRustCallAsync<F, T>(
+    rustFutureFunc: () -> UnsafeMutableRawPointer,
+    pollFunc: (UnsafeMutableRawPointer, UnsafeMutableRawPointer) -> (),
+    completeFunc: (UnsafeMutableRawPointer, UnsafeMutablePointer<RustCallStatus>) -> F,
+    freeFunc: (UnsafeMutableRawPointer) -> (),
+    liftFunc: (F) throws -> T,
+    errorHandler: ((RustBuffer) throws -> Error)?
+) async throws -> T {
+    // Make sure to call uniffiEnsureInitialized() since future creation doesn't have a
+    // RustCallStatus param, so doesn't use makeRustCall()
+    uniffiEnsureInitialized()
+    let rustFuture = rustFutureFunc()
+    defer {
+        freeFunc(rustFuture)
+    }
+    var pollResult: Int8;
+    repeat {
+        pollResult = await withUnsafeContinuation {
+            pollFunc(rustFuture, ContinuationHolder($0).toOpaque())
+        }
+    } while pollResult != UNIFFI_RUST_FUTURE_POLL_READY
+
+    return try liftFunc(makeRustCall(
+        { completeFunc(rustFuture, $0) },
+        errorHandler: errorHandler
+    ))
```

Comment:
**Suggestion:** Consider removing the force unwrapping in the `uniffiRustCallAsync` function to avoid potential crashes. [best practice]
```suggestion
    do {
        return try liftFunc(makeRustCall(
            { completeFunc(rustFuture, $0) },
            errorHandler: errorHandler
        ))
    } catch {
        print("Error in uniffiRustCallAsync: \(error)")
    }
```